be/src/olap/rowset/alpha_rowset_reader.cpp

```diff
@@ -194,11 +192,21 @@ OLAPStatus AlphaRowsetReader::_get_next_row_for_singleton_rowset(RowCursor** row
         }
     }
     if (min_row == nullptr || min_index == -1) {
-        return OLAP_ERR_READER_READING_ERROR;
+        return OLAP_ERR_DATA_EOF;
     }
     *row = min_row;
     RowBlock* row_block = _row_blocks[min_index];
```

Comment:
_row_blocks[min_index]->pos_inc() may be better

Reply:
ok